src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,94 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Movie = {
+  title: string,
+  description: string,
+  imgUrl: string,
+  imdbUrl: string,
+  imdbId: string,
+};
+
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [inputTitle, setInputTitle] = useState('');
+  const [inputDescription, setInputDescription] = useState('');
+  const [inputImage, setInputImage] = useState('');
+  const [inputUrl, setInputUrl] = useState('');
+  const [inputId, setInputId] = useState('');
+
+  const isValid = inputTitle && inputImage && inputUrl && inputId;
+
+  const movie = {
+    title: inputTitle,
+    description: inputDescription,
+    imgUrl: inputImage,
+    imdbUrl: inputUrl,
+    imdbId: inputId,
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    onAdd(movie);
+    setCount(currCount => currCount + 1);
+    setInputTitle('');
+    setInputDescription('');
+    setInputImage('');
+    setInputUrl('');
+    setInputId('');
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={(event) => {
+        handleSubmit(event);
+      }}
```

Comment:
no extra wrap needed

```suggestion
      onSubmit={handleSubmit}
```